### PR TITLE
Adtelligent support both userSync types

### PR DESCRIPTION
**Pull request by GeneGenie**
## Type of change
- [x] Feature

## Description of change
Adapter response may provide sync type.

- g.moroz@adtelligent.com
- [x] official adapter submission

PS: Tests updated

### CODE REVIEW

**Comment by GeneGenie** [comment]
@bretg Yes, your comment makes perfect sense. I think this change was requested to avoid pre-integration work with publishers. 
Anyway,  I will change the code.

File: test/spec/modules/adtelligentBidAdapter_spec.js [hunk 1]
```diff
@@ -109,17 +124,39 @@ const displayEqResponse = [{
...
+    it('should be returned if pixel enabled', function () {
+      const syncs = spec.getUserSyncs({pixelEnabled: true}, [{body: SERVER_DISPLAY_RESPONSE_WITH_MIXED_SYNCS}]);
+
+      expect(syncs.map(s => s.url)).to.deep.equal([SERVER_DISPLAY_RESPONSE_WITH_MIXED_SYNCS.cookieURLs[0]]);
+      expect(syncs.map(s => s.type)).to.deep.equal(['image']);
+    })
+  })
+
+  describe('user syncs as iframe', function () {
+    it('should be returned if pixel enabled', function () {
```
Comment (bretg) [inline test/spec/modules/adtelligentBidAdapter_spec.js]: I think you mean 'should be returned if iframe enabled'

File: test/spec/modules/adtelligentBidAdapter_spec.js [hunk 2]
```diff
@@ -109,17 +124,39 @@ const displayEqResponse = [{
...
+  describe('user syncs with both types', function () {
     it('should be returned if pixel enabled', function () {
-      const syncs = spec.getUserSyncs({pixelEnabled: true}, [{body: SERVER_DISPLAY_RESPONSE}]);
+      const syncs = spec.getUserSyncs({
+        iframeEnabled: true,
+        pixelEnabled: true
+      }, [{body: SERVER_DISPLAY_RESPONSE_WITH_MIXED_SYNCS}]);
 
-      expect(syncs.map(s => s.url)).to.deep.equal(SERVER_DISPLAY_RESPONSE.cookieURLs);
+      expect(syncs.map(s => s.url)).to.deep.equal(SERVER_DISPLAY_RESPONSE_WITH_MIXED_SYNCS.cookieURLs);
```
Comment (bretg) [inline test/spec/modules/adtelligentBidAdapter_spec.js]: So again - the original idea behind usersync was that iframes would be used if they were allowed by the pub and that image syncs were a fallback. We wouldn't expect _both_ types of syncs if both are allowed. Just iframe. Trying to minimize the number of syncs here.

Comment (GeneGenie) [reply test/spec/modules/adtelligentBidAdapter_spec.js]: Sorry. Did not get your point.

Comment (GeneGenie) [reply test/spec/modules/adtelligentBidAdapter_spec.js]: if two syncs are required, and everything is allowed, then 2 syncs are called as expected. Our lineItems may require triggering multiple syncs for single bid, and their types do not matter.

**Comment by GeneGenie** [comment]
Yieldbot test failing CI here.

**Comment by bretg** [comment]
Note that this is only in the master (2.x) branch right now. If you need it in the 1.40.x-legacy branch, will need to be copied there.

**Comment by GeneGenie** [comment]
@bretg yeaaaah, I also need this change in 1.40

**Pull request merged**